dspace-api/src/main/java/org/dspace/authorize/dao/impl/ResourcePolicyDAOImpl.java

```diff
@@ -153,16 +156,30 @@ public List<ResourcePolicy> findByTypeGroupActionExceptId(Context context, DSpac
 
     public List<ResourcePolicy> findByEPersonGroupTypeIdAction(Context context, EPerson e, List<Group> groups,
                                                                int action, int type_id) throws SQLException {
+        // If groups and eperson are empty, return immediately
+        if (CollectionUtils.isEmpty(groups) && e == null) {
+            return Collections.emptyList();
+        }
+
```

Comment:
I was attempting to model the existing behavior here, as this is another example of code where Hibernate 6 is much less forgiving about null values.  In Hibernate 5, if the Groups & Eperson were empty, this method would *still run* but would always return no results.  In Hibernate 6, if these objects are null/empty, then the method was throwing a Hibernate Syntax error when the query was run.  That's why I had to modify this method to ensure we are not passing null or empty values to Hibernate.  

So, this code is attempting to ensure we keep the same behavior as before... which is why an empty set of results is returned instead of an error.

I'm not against changing the behavior to throw an exception, but that would likely require additional code changes to methods which call this one.  I think that's out of scope for this PR, but it's something we could do later on if we felt it was necessary.

Reply:
Again we should not clank the database when we *know* there will be no result; we should just return no result.  I wonder whether we should log a warning, but will not argue for it at this time.

We should probably follow up with a careful examination of our database code to comb out such places where we are lazy and lean on the DBMS to tell us that we asked for nothing.  AFTER this patch goes in.

I agree with the change in Hibernate.  Java `null` does not necessarily mean SQL `IS NULL`.  If you are searching for rows with null column values you should be required to explicitly say so.  The old behavior was probably the source of many mysterious errors.

Reply:
Yes, I agree that I actually *prefer* the new Hibernate behavior.  It forces us to filter out the null values *before* using Hibernate to query anything.  But, it did require some code cleanup in this PR to add those extra null/empty checks. I **think** I caught them all...but I agree we should keep an eye on our database queries better after this is merged.

@mwoodiupui : Do you want a chance to still check this PR further, or is it good to merge?  (I'd like to merge today/tomorrow if at all possible)